api/api/clubs/tests.py

```diff
@@ -55,3 +56,55 @@ def testListClubs_hundredClubExistsInDb_returnsFiftyClubInFirstResponse(self):
         response = self.client.get('/clubs/', format='json')
         # then
         self.assertEqual(len(response.data['results']), 50)
+
+
+class UpdateClubsViewTest(APITestCase):
+    """Tests UpdateClubVew for updating club details """
+
+    @pytest.mark.django_db
+    def testUpdateClubs_nonAdminRequest_throwsForbidden(self):
+        # given
+        user = User.objects.create(
+            email='test@user.com', first_name='Test', last_name='User')
+        self.client.force_authenticate(user)
+
+        test_club = Club.objects.create(name="Bitbyte - The Programming Club",
+                                        category="S&T",
+                                        description="Some desc here",
+                                        email="theprogclub@iiitdmj.ac.in",
+                                        logo="#")
+        # when
+        response = self.client.put(f'/clubs/{test_club.id}/', {'name': 'Bitbyte',
+                                                               'category': "S&T",
+                                                               'description': "Some desc here",
+                                                               'email': "theprogclub@iiitdmj.ac.in",
+                                                               'logo': "https://iiitdmj.ac.in"})
+
+        # then
+        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
+
+    @pytest.mark.django_db
+    def testUpdateClubs_AdminRequest_updatesSuccessful(self):
+        # given
+        user = User.objects.create(
+            email='test@user.com', first_name='Test', last_name='User', is_staff=True)
+        self.client.force_authenticate(user)
+
+        test_club = Club.objects.create(name="Bitbyte - The Programming Club",
+                                        category="S&T",
+                                        description="Some desc here",
+                                        email="theprogclub@iiitdmj.ac.in",
+                                        logo="#")
+        # when
+        response = self.client.put(f'/clubs/{test_club.id}/', {'name': 'changed_club_name',
+                                                               'category': "S&T",
+                                                               'description': "Some desc here",
+                                                               'email': "theprogclub@iiitdmj.ac.in",
+                                                               'logo': "https://iiitdmj.ac.in"})
+
+        # then
+        updated_club = Club.objects.get(id=test_club.id)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(response.data['name'], 'changed_club_name')
+        self.assertEqual(updated_club.name, 'changed_club_name')
+        self.assertEqual(updated_club.category, 'S&T')
```

Comment:
if `serialization.is_valid` fails a `BadRequestError` needs to be thrown . Can add `test` for it